### DISCUSSION TITLE
Sync mobile devices - download changed/new files only

### DISCUSSION THREAD

**Discussion by alpinfra** [opengisch/QField#3394] [Q&A]
I have set up a QField job with two collaborators (editors). 
When editing the project on a mobile device (files are stored in a GeoPackage), I can upload the changed files including pictures to the cloud. Fine.
But I can only get the updated data on the second device, if I synchronize the entire project, which takes some time and is unnecessary (regarding bandwidth and idle time). The option "refresh projects" does not download the changed data. Is there any option that I may have missed on a mobile device to update changed and download new files only? I'm only working on Android mobile devices by the way.

When working on a desktop, I can choose via the QFieldCloud-plugin, whether I want to download the changed files -> perfect.
It may only take a couple of minutes until the changes uploaded from a mobile device get the status applied on the cloud server.

Help greatly appreciated
regards

**Comment by suricactus** [comment]
Hi @alpinfra . Note that even if you make a very small change in your Geopackage file, it is the whole file that changed, therefore we package the whole project to make sure all the dependencies are met.

There is room for improvement to speed up packaging for projects with only GeoPackages, but it is not in the immediate priority list. 

Feel free to drop us an email if you want to speed up the development of such a feature.

**Reply by alpinfra** [reply]
I understand your intention, but I would very much welcome the option only to "get changes" of the project, just like you are able to "push changes".
